direntry/direntry.go

```
@@ -0,0 +1,120 @@
+package direntry
+
+import (
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+)
+
+var ignoreDirs = []string{".git", ".github", ".gradle", ".idea", "build", ".kotlin", ".fleet", "CordovaLib", "node_modules"}
+
+type DirEntry struct {
```

I see you wrote your own recursive filesystem directory walking logic. Have you thought simply wrapping the default `filepath.Walk`? I think it can do the same thing as what you are doing here.

If you need to skip a dir then you can return a special error:
```
err = filepath.Walk(path/to/dir, func(path string, info fs.FileInfo, err error) error {
	if iNeedToSkipThis {
		return filepath.SkipDir
	}
	...
})
```

If you need to limit the depth then you can check the returned path for how many components it has:
```
err = filepath.Walk(path/to/dir, func(path string, info fs.FileInfo, err error) error {
	if strings.Count(path, string(os.PathSeparator)) > maxDepth {
		return filepath.SkipDir
	}
	...
})
```

And if you want to find a single thing and terminate the execution then you can return another special error:
```
err = filepath.Walk(path/to/dir, func(path string, info fs.FileInfo, err error) error {
	if iNeedToStop {
		return filepath.SkipAll
	}
	...
})
```

---

We discussed keeping the current directory parsing logic because filepath.Walk would make it harder to build relationship between directories and its parent entries.